code/obj/sawflymisc.dm

```diff
@@ -0,0 +1,581 @@
+
+	health = 20 // ignore this, they have 40 hp.
+	var/can_revive = TRUE
+	var/atksilicon = TRUE
+	var/atkcarbon = TRUE
+	var/alive = TRUE
+
+// OBJ/CRITTER/DRONE FUCK SHIT HERE
+	var/smashes_shit = 0
+	var/list/friends = list()
+	var/attacking = 0
+	var/atk_delay = 5
+	var/attack_cooldown = 10
+	var/seekrange = 15
+	var/attacker = null
+	var/atom/movable/target = null
+	var/oldtarget_name = null
+	var/attack = 0
+	var/task = "thinking"
+	var/frustration = 0
+	var/atom/target_lastloc = null
+	var/aggressive = 1
+	var/last_found = null
+	var/steps = 0
+	var/sleep_check = 10
+	var/wander_check = 0
+
+
+	// MOB/LIVING/CRITTER FUCK SHIT HERE
+	setup_healths()
+		add_hh_robot(20, 1) // fuck you and your way of doing health that isn't my way of doing it
+		add_hh_robot_burn(20, 1)
+	custom_gib_handler = /proc/robogibs
+	blood_id = "oil"
+	use_stamina = FALSE
+	use_stunned_icon = FALSE
+	butcherable = FALSE
+	can_bleed = FALSE
+	canbegrabbed = FALSE
+	can_lie = FALSE
+	can_burn = FALSE
+	pet_text = "cuddles"
+
+// special noises so they can't be confused with robots
+	var/beep1 = 'sound/machines/sawfly1.ogg'
+	var/beep2 = 'sound/machines/sawfly2.ogg'
+	var/beep3 = 'sound/machines/sawfly3.ogg'
+
+
+	New()
+		..()
+		if(isnew)
+			name = "Sawfly [pick(sawflynames)]-[rand(1,999)]"
+		deathtimer = rand(1, 5)
+		death_text = "[src] jutters and falls from the air, whirring to a stop."
+		beeptext = "[pick(list("beeps",  "boops", "bwoops", "bips", "bwips", "bops", "chirps", "whirrs", "pings", "purrs", "thrums"))]"
+		animate_bumble(src) // gotta get the float goin' on
+		src.set_a_intent(INTENT_HARM) // incredibly stupid way of ensuring they aren't passable
+
+
+	proc/foldself()
+		var/obj/item/old_grenade/spawner/sawfly/reused/N = new /obj/item/old_grenade/spawner/sawfly/reused(get_turf(src))
+		// pass our name and health
+		N.name = "Compact [name]"
+		N.tempname = src.name
+		N.temphp = (src.get_health_percentage()) / 2
+		qdel(src)
+
+	proc/randombeep() //adds some variety into the noises they make
+		var/beepnum=rand(1, 3)
+		if (beepnum == 1)
+			playsound(src, beep1, 55, 1)
+		else if (beepnum == 2)
+			playsound(src, beep2, 55, 1)
+		else
+			playsound(src, beep3, 55, 1)
+
+
+
+	emp_act() //same thing as if you emagged the controller, but much higher chance
+		if(prob(80))
+			src.visible_message("<span class='combat'>[src] buzzes oddly and starts to sprial out of contro!</span>")
+			SPAWN(2 SECONDS)
+				src.blowup()
+		else
+			src.foldself()
+
+
+	proc/CritterDeath() // rip lil guy
+
+		if (!src.alive) return // can't double-die
+		alive = FALSE
+		src.tokenized_message(death_text)
+		src.is_npc = FALSE // stop any and all possible non-critter AI thought
+		SEND_GLOBAL_SIGNAL(COMSIG_GLOBAL_DRONE_DEATH, src)
+
+		if (prob(20))
+			new /obj/item/device/prox_sensor(src.loc) // maybe change this later
+			return
+
+
+		/*SHOULD_CALL_PARENT(TRUE) //commenting this out because I'm afraid everything will break if I delete it
```

Comment:
This should be deleted, you don't want to send the signal for obj critter death to components attached to a mob critter. That would be bad.